README.md

```diff
@@ -148,3 +148,67 @@ set the logger level to INFO:
 
     logging.getLogger('backoff').setLevel(logging.INFO)
 
+## Examples for Full Jitter and Equal Jitter
```

Comment:
Two things:

1. Currently jitter isn't documented in the README at all (although it is explained in the docstrings for the on_exception and on_predicate functions). Now that full_jitter will be the default, I think this little bit of docs should really be called *Jitter* and be a subsection of the main Examples heading. I don't want to overly nitpick the docs and prevent this PR from getting in though. If you prefer to temporarily remove this section for now that's fine too. It's something I can revise as we go on in subsequent commits.
2. I didn't mention this before because I didn't want to complicate the review, but currently the markdown docs are actually defined in the docstring for backoff.py.  Using the `docs` Makefile target (`make docs`) generates the README.md file. The reason is I originally wanted the same documentation available via `help(backoff)` in interactive python. However, I think the docs have pretty clearly outgrown this now and there is more that we want in the README than we need in the docstring.  I will work on changing this, so it's probably fine to continue to add the new docs directly to the README, and I'll work on simplifying the docstring so that it doesn't try to cover everything.